sys/linux/auto.txt

```diff
@@ -1059,63 +1039,271 @@ syz_genetlink_get_family_id$auto_tcp_metrics(name ptr[in, string["tcp_metrics"]]
+	esterror		intptr
+	status			int32
+	__kernel_timex_7	int32
+	constant		intptr
+	precision		intptr
+	tolerance		intptr
+	time			__kernel_timex_timeval$auto_record
+	tick			intptr
+	ppsfreq			intptr
+	jitter			intptr
+	shift			int32
+	__kernel_timex_16	int32
+	stabil			intptr
+	jitcnt			intptr
+	calcnt			intptr
+	errcnt			intptr
+	stbcnt			intptr
+	tai			int32
+	__kernel_timex_23	int32
+	__kernel_timex_24	int32
+	__kernel_timex_25	int32
+	__kernel_timex_26	int32
+	__kernel_timex_27	int32
+	__kernel_timex_28	int32
+	__kernel_timex_29	int32
+	__kernel_timex_30	int32
+	__kernel_timex_31	int32
+	__kernel_timex_32	int32
+	__kernel_timex_33	int32
+}
+
+__kernel_timex_timeval$auto_record {
+	tv_sec	intptr
+	tv_usec	intptr
+}
+
+__old_kernel_stat$auto_record {
+	st_dev		int16
+	st_ino		int16
+	st_mode		int16
+	st_nlink	int16
+	st_uid		int16
+	st_gid		int16
+	st_rdev		int16
+	st_size		int32
+	st_atime	int32
+	st_mtime	int32
+	st_ctime	int32
+}
+
+__sifields$auto_record [
+	_kill		_kill$auto_record
+	_timer		_timer$auto_record
+	_rt		_rt$auto_record
+	_sigchld	_sigchld$auto_record
+	_sigfault	_sigfault$auto_record
+	_sigpoll	_sigpoll$auto_record
+	_sigsys		_sigsys$auto_record
+]
+
+__user_cap_data_struct$auto_record {
+	effective	int32
+	permitted	int32
+	inheritable	int32
+}
+
+__user_cap_header_struct$auto_record {
+	version	int32
+	pid	int32
+}
+
+_addr_bnd$auto_record {
+	_dummy_bnd	array[int8, 8]
+	_lower		ptr[inout, void]
+	_upper		ptr[inout, void]
+}
+
+_addr_pkey$auto_record {
+	_dummy_pkey	array[int8, 8]
+	_pkey		int32
+}
+
+_kill$auto_record {
+	_pid	int32
+	_uid	int32
+}
+
+_perf$auto_record {
+	_data	int32
+	_type	int32
+	_flags	int32
+}
+
+_rt$auto_record {
+	_pid	int32
+	_uid	int32
+	_sigval	sigval$auto_record
+}
+
+_sigchld$auto_record {
+	_pid	int32
+	_uid	int32
+	_status	int32
+	_utime	int32
+	_stime	int32
+}
+
+_sigev_thread$auto_record {
+	_function	ptr[inout, ptr[in, int8]]
+	_attribute	ptr[inout, void]
+}
+
+_sigev_un$auto_record [
+	_pad		array[int32, 12]
+	_tid		int32
+	_sigev_thread	_sigev_thread$auto_record
+]
+
+_sigfault$auto_record {
+	_addr		ptr[inout, void]
+	_sigfault_1	_sigfault_1$auto_record
+}
+
+_sigfault_1$auto_record [
+	_trapno		int32
+	_addr_lsb	int16
+	_addr_bnd	_addr_bnd$auto_record
+	_addr_pkey	_addr_pkey$auto_record
+	_perf		_perf$auto_record
+]
+
+_sigpoll$auto_record {
+	_band	int32
+	_fd	int32
+}
+
+_sigsys$auto_record {
+	_call_addr	ptr[inout, void]
+	_syscall	int32
+	_arch		int32
+}
+
+_timer$auto_record {
+	_tid		int32
+	_overrun	int32
+	_sigval		sigval$auto_record
+	_sys_private	int32
+}
+
 ack_mapinfo_policy$auto_iwpm_msg [
 	IWPM_NLA_MAPINFO_SEQ		nlattr[IWPM_NLA_MAPINFO_SEQ, int32]
 	IWPM_NLA_MAPINFO_SEND_NUM	nlattr[IWPM_NLA_MAPINFO_SEND_NUM, int32]
 	IWPM_NLA_MAPINFO_ACK_NUM	nlattr[IWPM_NLA_MAPINFO_ACK_NUM, int32]
 ] [varlen]
 
 act_bpf_policy$auto_act_bpf [
-	TCA_ACT_BPF_PARMS	nlattr[TCA_ACT_BPF_PARMS, array[int8, 20]]
+	TCA_ACT_BPF_PARMS	nlattr[TCA_ACT_BPF_PARMS, array[tc_act_bpf$auto_record]]
```

Comment:
This is declared as:
```
	[TCA_ACT_BPF_PARMS]	= { .len = sizeof(struct tc_act_bpf) },
````
https://elixir.bootlin.com/linux/v6.11/source/net/sched/act_bpf.c#L178

If max size is equal to the size of a single struct, it shouldn't be an array.